metis/lib/file_rename_revision.rb

```diff
@@ -0,0 +1,72 @@
+require_relative 'revision'
+
+class Metis
+  class FileRenameRevision < Revision
+    def validate
+      @errors = []
+      validate_source(@source)
+      validate_dest(@dest)
+    end
+
+    def bucket_names
+      # This is kind of weird, but we need the ability to grab
```

Comment:
i find this method difficult to follow  - might you instead push @source and @dest into @buckets in the initializer, then have bucket_names just iterate across @buckets in `Revision`? This would save having to update this method whenever you subclass...